### PR TITLE
Remove legacy gov proposal dependencies

### LINKED ISSUE #1583: Remove v1beta1 gov proposals integration
- [x] Remove integration in app.go
~- [ ] Remove deprecated code from the repo~ moved to #1608

### PR DESCRIPTION
**Pull request by pinosu**
Resolves #1583

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -483,16 +453,6 @@ func NewWasmApp(
 		scopedIBCKeeper,
 	)
 
-	// Register the proposal types
-	// Deprecated: Avoid adding new handlers, instead use the new proposal flow
-	// by granting the governance module the right to execute the message.
-	// See: https://docs.cosmos.network/main/modules/gov#proposal-messages
-	govRouter := govv1beta1.NewRouter()
```
Comment (alpe) [inline app/app.go:490]: The legacy router is still used for some modules. See https://github.com/cosmos/cosmos-sdk/blob/v0.47.4/simapp/app.go#L357-L360

File: x/wasm/keeper/test_common.go [hunk 2]
```diff
@@ -452,11 +450,6 @@ func createTestInput(
 	// add wasm handler so we can loop-back (contracts calling contracts)
 	contractKeeper := NewDefaultPermissionKeeper(&keeper)
 
-	govRouter := govv1beta1.NewRouter().
```
Comment (alpe) [inline x/wasm/keeper/test_common.go:455]: no need for beta1 here 👍

File: x/wasm/keeper/proposal_integration_test.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ import (
 const myTestLabel = "testing"
 
 func TestStoreCodeProposal(t *testing.T) {
+	t.Skip("DEPRECATED")
```
Comment (alpe) [inline x/wasm/keeper/proposal_integration_test.go:28]: I think it is fine to just remove the tests, now

File: README.md [hunk 4]
```diff
@@ -198,9 +198,6 @@ Available flags:
  
 * `-X github.com/CosmWasm/wasmd/app.NodeDir=.corald` - set the config/data directory for the node (default `~/.wasmd`)
 * `-X github.com/CosmWasm/wasmd/app.Bech32Prefix=coral` - set the bech32 prefix for all accounts (default `wasm`)
-* `-X github.com/CosmWasm/wasmd/app.ProposalsEnabled=true` - enable all x/wasm governance proposals (default `false`)
```
Comment (alpe) [inline README.md:201]: very good to keep the Readme updated 👍

**Review by alpe — COMMENTED** [review]
Very nice! Just make sure that you do not fully remove the legacy router. It may still be needed by some modules/ in flight proposals

**Review by alpe — APPROVED** [review]
The changes look good now! Thanks for taking care!
The README contains a section about [runtime flags](https://github.com/CosmWasm/wasmd#runtime-flags) that should be updated, too. Please remove the gov example there.

**Comment by pinosu** [comment]
@alpe thanks for the review. I think I have removed everything from the README already. 
I only found the one example in [Vote on the upgrade](https://github.com/CosmWasm/wasmd/blob/7bd656690b3f1d60598b1f31e3d95571fbf1bac8/UPGRADING.md?plain=1#L167) and some more docs in [Governance.md](https://github.com/CosmWasm/wasmd/blob/7bd656690b3f1d60598b1f31e3d95571fbf1bac8/x/wasm/Governance.md?plain=1#L37) . Should I update also these?

**Comment by alpe** [comment]
> I only found the one example in [Vote on the upgrade](https://github.com/CosmWasm/wasmd/blob/7bd656690b3f1d60598b1f31e3d95571fbf1bac8/UPGRADING.md?plain=1#L167) and some more docs in [Governance.md](https://github.com/CosmWasm/wasmd/blob/7bd656690b3f1d60598b1f31e3d95571fbf1bac8/x/wasm/Governance.md?plain=1#L37) . Should I update also these?

It makes sense to have the other docs updated, too. 👍

**Pull request merged**